calculator.js

```diff
@@ -0,0 +1,142 @@
+let state = {
+  currentOperation: null, // это может быть сложение, умножение, вычитание, etc. Ключи: -, + и тд
+  immediateResult: 0, // "мгновенный" результат. Используется только для рендеринга, !не для расчетов!
+  fixedResult: 0, // используется как опорное значение. На его основе считается immediateResult
+  currentOperand: "", // это буфер, хранящий текущее число. Чтоб не плюсовали 1, потом 2, потом 3, а сразу 123
+};
+
+const setState = (key, payload) => {
+  state = { ...state, [key]: payload };
+  renderImmediateResult();
+  console.log(state);
+};
+
+function setInitialState() {
+  setState("currentOperation", null);
+  setState("immediateResult", 0);
+  setState("fixedResult", 0);
+  setState("currentOperand", "");
+}
+
+const DIGIT_BUTTONS = document.querySelectorAll(".digit");
+const OPERATOR_BUTTONS = document.querySelectorAll(".operation");
+const BTN_VALUES = Array.from(document.querySelectorAll(".button")).map(
+  (el) => el.innerHTML
+);
+const KEYBOARD = document.querySelector(".keyboard");
+const INPUT = document.querySelector("#myInput");
+const ALL_CLEAR = document.querySelector(".ac");
+const PLUS_BUTTON = document.querySelector(".plus");
+const RESULT_SCREEN = document.querySelector(".screen-number");
+const EQUAL_BUTTON = document.querySelector(".equal");
+const SCREEN_EQUAL = document.querySelector(".screen-equal");
+const MULTIPLY_BUTTON = document.querySelector(".multiply");
+const DIVISION_BUTTON = document.querySelector(".division");
+const MINUS_BUTTON = document.querySelector(".minus");
+
+KEYBOARD.addEventListener("click", handleKeyboardClick);
+ALL_CLEAR.addEventListener("click", handleAllClearClick);
+EQUAL_BUTTON.addEventListener("click", handleEqualClick);
+EQUAL_BUTTON.addEventListener("click", renderImmediateResult);
+PLUS_BUTTON.addEventListener("click", handlePlusClick);
+MULTIPLY_BUTTON.addEventListener("click", handleMultiplyClick);
+DIVISION_BUTTON.addEventListener("click", handleDivisionClick);
+MINUS_BUTTON.addEventListener("click", handleMinusClick);
+DIGIT_BUTTONS.forEach((btn) => btn.addEventListener("click", handleDigitClick));
+OPERATOR_BUTTONS.forEach((btn) => {
+  btn.addEventListener("click", (e) => {
+    if (state.currentOperation) {
+      setState("fixedResult", state.immediateResult);
+    }
+  });
+});
+
+function handleKeyboardClick(event) {
+  const btnValue = event.target.innerHTML;
+  if (!BTN_VALUES.includes(btnValue) || btnValue === "AC") {
+    return;
+  }
+  INPUT.value = INPUT.value + btnValue;
+}
+
+function handleAllClearClick() {
+  setInitialState();
+  INPUT.value = "";
+  RESULT_SCREEN.innerHTML = 0;
```

Comment:
imo, both lines should be in render() function (I left more details below)

Reply:
done